engine/src/test/java/io/camunda/zeebe/engine/processing/deployment/model/validation/TimerValidationTest.java

```diff
@@ -177,6 +188,16 @@ private static Stream<Arguments> timerEventsWithExpression() {
                             .intermediateCatchEvent("intermediate-catch-event")
                             .timerWithDurationExpression(expression)
                             .done())),
+            Arguments.of(
+                "intermediate catch event",
+                "date",
+                processBuilder(
+                    expression ->
+                        Bpmn.createExecutableProcess("process")
+                            .startEvent()
+                            .intermediateCatchEvent("intermediate-catch-event")
+                            .timerWithDateExpression(expression)
+                            .done())),
```

Comment:
❌  We should add a case for the event-sub process here as well.

Reply:
Thank you very much. I will try to do so. Hope you are feeling well :)

Reply:
@remcowesterhoud I have added subprocess validation tests and some timer tests. I have limited it to test regarding sub-processes started by a timer, assuming that generic timer or generic sub-processes concerns are tested in those aspects.